torch/package/package_importer.py

```diff
@@ -340,17 +357,26 @@ def _make_module(
     def _load_module(self, name: str, parent: str):
         cur: _PathNode = self.root
         for atom in name.split("."):
-            if not isinstance(cur, _PackageNode) or atom not in cur.children:
+            if isinstance(cur, _SelectiveInternNode):
+                if atom not in cur.children:
+                    cur.children[atom] = _ExternNode()
+            elif not isinstance(cur, _PackageNode) or atom not in cur.children:
                 raise ModuleNotFoundError(
                     f'No module named "{name}" in self-contained archive "{self.filename}"'
                     f" and the module is also not in the list of allowed external modules: {self.extern_modules}",
                     name=name,
                 )
             cur = cur.children[atom]
-            if isinstance(cur, _ExternNode):
+            if isinstance(cur, (_ExternNode, _SelectiveExternNode)):
```

Comment:
I'm also confused by this. If `torch` is a selective externed node, we don't want to load `torch` from `importlib`, which will give us the "actually externed" version of `torch`.